src/dataflow/components/dataflow-program-graph.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from "react";
+import { Line } from "react-chartjs-2";
+import { ChartOptions, ChartData, ChartDataSets } from "chart.js";
+import { MAX_NODE_VALUES } from "./dataflow-program";
+import { NodePlotColors } from "./../utilities/node";
+import "./dataflow-program-graph.sass";
+
+export interface DataPoint {
+    x: number;
+    y: number;
+}
+export interface DataSequence {
+  name: string;
+  units: string;
+  data: DataPoint[];
+}
+export interface DataSet {
+  sequences: DataSequence[];
+}
+
+interface IProps {
+  dataSet: DataSet;
+}
+interface IState {
+  stacked: boolean;
+  scatter: boolean;
+}
+
+export class DataflowProgramGraph extends React.Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      stacked: true,
+      scatter: true,
+    };
+  }
+
+  public handleLayoutClick = () => {
+    const stacked = !this.state.stacked;
+    this.setState({stacked});
+  }
+  public handleTypeClick = () => {
+    const scatter = !this.state.scatter;
+    this.setState({scatter});
+  }
+
+  public render() {
+    return (
+      <div className="program-graph" data-test="program-graph">
+        { this.state.stacked
+          ? this.renderStackedGraphs()
+          : this.renderOverlappedGraphs()
+        }
+        <button className="graph-button layout" onClick={this.handleLayoutClick}>
+          { this.state.stacked ? "overlap" : "stack" }
+        </button>
+        <button className="graph-button type" onClick={this.handleTypeClick}>
+          { this.state.scatter ? "line" : "scatter" }
+        </button>
```

Comment:
These buttons and associated functionality might get pulled in the final iteration, but were desired so people can weigh which version of the graphed data is preferred.